### ISSUE TITLE
Could you clarify the requirements for rasp.Aggregate?

### ISSUE THREAD

**Issue by langosco** [google-deepmind/tracr#17]
In compiler/validating.py, a categorical rasp.Aggregate fails validation if (in DynamicValidationEvaluator)
```python
sel_width_val > 1 and set(agg_in) != set(out)
```
That is, if the selector has width > 1 and the input domain does not match the output domain.

However, rasp/rasp.py contains another condition that triggers when the rasp expression is called (in eval_aggregate):
```python
  exemplar = xs[0]
  if isinstance(exemplar, (int, bool)):
    return sum(xs) / len(xs)
  elif len(xs) == 1:
    return exemplar
  else:
    raise ValueError(f"Unsupported type for aggregation: {xs}")
```
That is, rasp.Aggregate only supports int or bool inputs whenever selector width > 1, which is an additional criterion not present in the validator. For example, if `xs=[0.5, 1.0, 1.5]` and the selector contains the row `[1, 0, 1]`, then the output corresponding to that row is 1.0, which is present in the input SOp (and so can satisfy the validation requirement).

I don't know what the actual requirements for the compiler are, but I would have expected that the python type of the input doesn't matter as long as all the SOps involved are categorical. Is that right?

**Comment by langosco** [comment]
### Additional questions
Can the condition in the validator
```python
sel_width_val > 1 and set(agg_in) != set(out)
```
be weakened (or strengthened depending on how one sees it) to:
```python
sel_width_val > 1 and not set(out).issubset(set(agg_in))
```

**Comment by david-lindner** [comment]
> I would have expected that the python type of the input doesn't matter as long as all the SOps involved are categorical. Is that right?

Yes you are right, categorical SOps should always be supported. The check in the evaluation should only check for which types the average operation is well defined, so it should definitely also allow floats. Feel free to send a PR to add float there.


> Can the condition in the validator be weakened

Yes I think that should also work and would be better. Again, feel free to send a PR (if you haven't seen any examples that contradict this)

**Comment by langosco** [comment]
Great! Done in PR #18

**Comment by david-lindner** [comment]
Thanks! Merged #18
